controllers/provider-azure/pkg/internal/infrastructure/terraform.go

```diff
@@ -277,6 +307,16 @@ func StatusFromTerraformState(state *TerraformState) *apiv1alpha1.Infrastructure
 		tfState.Networks.VNet.ResourceGroup = &state.VNetResourceGroupName
 	}
 
+	if state.IdentityID != "" && state.IdentityClientID != "" {
+		tfState.Identity = &apiv1alpha1.IdentityStatus{
+			ID:       state.IdentityID,
+			ClientID: state.IdentityClientID,
+		}
+		if config.Identity != nil && config.Identity.AcrAccess != nil && *config.Identity.AcrAccess {
+			tfState.Identity.AcrAccess = true
```

Comment:
Hm, this part looks a little weird. You are setting same field in the `tfState` object based on the `config`. This function was only evaluation the terraform state and computing the `InfrastructureStatus` out of it. Now it's mixed with `InfrastructureConfig`. I would clean this up.